src/qt/window.cpp

```diff
@@ -193,6 +193,29 @@ static const char WINDOW_POINTER_PROPERTY_NAME[] = "wxWindowPointer";
     return const_cast< wxWindowQt * >( ( variant.value< const wxWindow * >() ));
 }
 
+/* static */ void wxWindowQt::SendSetCursorEvent(wxWindowQt* win, wxPoint posScreen)
+{
+    wxWindowQt* w = win;
+    while( true )
+    {
+        const wxPoint posClient = w->ScreenToClient(posScreen);
+        wxSetCursorEvent event(posClient.x, posClient.y);
+        event.SetEventObject(w);
+
+        const bool processedEvtSetCursor = w->ProcessWindowEvent(event);
+        if ( processedEvtSetCursor && event.HasCursor() )
+        {
+            win->SetCursor(event.GetCursor());
+            return;
+        }
+
+        w = w->GetParent();
+        if ( w == NULL )
+            break;
+    }
+    win->SetCursor(wxCursor(wxCURSOR_ARROW));
```

Comment:
We did see some problems with this locally, with the wxSplitterWindow not showing a cursor when hovering over a splitter, which is set directly rather than using events. We tried wrapping the setCursor with a check on `if (!win->m_cursor.IsOk())`, which solved this but led to more situations where the wrong cursor would remain set when moving away from the relevant area. In the end we decided having the default cursor was better than showing a custom cursor in the wrong situation.

I think there is some more complexity in the GTK version where it tracks whether a reset of the mouse cursor is required but we couldn't quite manage to get it working the same way.

Reply:
See #24217